File: JavaScript/9.js
function generateData(n){
    var data = [];
    for(var i=0;i<n;i++){
      data.push(Math.floor(Math.random()*100)+1);
    }
    return data;
  }
  function computeStats(data){
    var sum = 0;
    for(var i=0;i<data.length;i++){
      sum += data[i];
    }
    var avg = data.length > 0 ? sum/data.length : 0;
    var min = Math.min.apply(null, data);
    var max = Math.max.apply(null, data);
    return {sum: sum, average: avg, min: min, max: max};
  }
  function filterData(data, threshold){
    var filtered = [];
    for(var i=0;i<data.length;i++){
      if(data[i] > threshold){
        filtered.push(data[i]);
      }
    }
    return filtered;
  }
  var data = generateData(30);
  console.log("Data: " + data);
  var stats = computeStats(data);
  console.log("Sum: " + stats.sum);
  console.log("Average: " + stats.average);
  console.log("Min: " + stats.min);
  console.log("Max: " + stats.max);
  var filtered = filterData(data, 50);
  console.log("Filtered (>50): " + filtered);
  for(var i=0;i<5;i++){
    console.log("Loop "+i);
  }
  for(var j=0;j<3;j++){
    for(var k=0;k<4;k++){
      console.log("Nested loop "+j+","+k);
    }
  }
  console.log("Data analysis simulation complete");
  var x = 0;
  while(x<4){
    console.log("While loop iteration "+x);
    x++;
  }
  for(var a=0;a<3;a++){
    console.log("Extra loop "+a);
  }
  for(var b=0;b<3;b++){
    for(var c=0;c<3;c++){
      console.log("Double nested "+b+","+c);
    }
  }
  console.log("Extended analysis start");
  var extData = generateData(15);
  console.log("Extended Data: " + extData);
  var extStats = computeStats(extData);
  console.log("Extended Sum: " + extStats.sum);
  console.log("Extended Average: " + extStats.average);
  console.log("Extended Min: " + extStats.min);
  console.log("Extended Max: " + extStats.max);
  console.log("Extra analysis details:");
  for(var m=0;m<4;m++){
    for(var n=0;n<3;n++){
      console.log("Extra nested "+m+","+n);
    }
  }
  console.log("Extended analysis complete");
  for(var i=0;i<3;i++){
    for(var j=0;j<3;j++){
      console.log("Final nested "+i+","+j);
    }
  }
  console.log("Data analysis simulation end");
  for(var p=0;p<3;p++){
    console.log("Closing loop "+p);
  }
  console.log("Simulation finished");
  for(var t=0;t<2;t++){
    console.log("Last loop "+t);
  }
  console.log("Goodbye from Data Analysis Simulation");
  console.log("Final end");
  

File: Python/9.py
import random
def generate_data(n):
 data=[]
 for i in range(n):
  data.append(random.randint(1,100))
 return data
def compute_stats(data):
 total=sum(data)
 count=len(data)
 average=total/count if count>0 else 0
 minimum=min(data) if data else None
 maximum=max(data) if data else None
 return total,average,minimum,maximum
def filter_data(data,threshold):
 filtered=[]
 for d in data:
  if d>threshold:
   filtered.append(d)
 return filtered
data=generate_data(20)
total,average,minimum,maximum=compute_stats(data)
print("Data:",data)
print("Total:",total)
print("Average:",average)
print("Min:",minimum)
print("Max:",maximum)
filtered=filter_data(data,50)
print("Filtered ( >50):",filtered)
for i in range(3):
 print("Loop",i)
for j in range(2):
 for k in range(3):
  print("Nested loop",j,k)
print("Data analysis simulation complete")
x=0
while x<3:
 print("While loop iteration",x)
 x+=1
for a in range(2):
 print("Extra loop",a)
for b in range(2):
 for c in range(2):
  print("Double nested",b,c)
print("Beginning extended analysis")
ext_data=generate_data(10)
print("Extended data:",ext_data)
sum_ext=sum(ext_data)
avg_ext=sum_ext/len(ext_data)
print("Extended Total:",sum_ext)
print("Extended Average:",avg_ext)
for m in range(3):
 for n in range(2):
  print("Extra nested",m,n)
print("Extended analysis complete")
def dummy_calc(x):
 if x%2==0:
  return x/2
 else:
  return x*3+1
result=[]
for num in data:
 result.append(dummy_calc(num))
print("Dummy calc results:",result)
for i in range(2):
 for j in range(2):
  print("Final nested",i,j)
print("Data analysis simulation end")
for i in range(2):
 print("Closing loop",i)
print("Simulation finished")
